app/Http/Controllers/Backend/BrouterController.php

```diff
@@ -0,0 +1,168 @@
+     * @param string $profile
+     *
+     * @return mixed|null
+     * @throws \JsonException
+     */
+    public static function getGeoJSONForRoute(
+        array  $coordinates,
+        string $profile = 'rail' //Maybe extend this for other travel types later
+    ): mixed {
+        $coordinateString = 'brouter?lonlats=';
+        foreach ($coordinates as $coords) {
+            $lonlat           = implode(',', array_reverse($coords)); //brouter needs order lon,lat
+            $coordinateString .= $lonlat . '|';
+        }
+        $coordinateString = rtrim($coordinateString, '|'); // Remove last "|"
+
+        $response = self::getHttpClient()
+                        ->get(strtr('brouter?lonlats=:coords&profile=:profile&alternativeidx=0&format=geojson', [
+                            ':coords'  => $coordinateString,
+                            ':profile' => $profile,
+                        ]));
+        Log::debug('Brouter URL is ' . $response->effectiveUri());
+        if (!$response->ok()) {
+            Log::debug('Brouter response was not okay.', ['body' => $response->body()]);
+            return null;
+        }
+
+        return json_decode($response->body(), false, 512, JSON_THROW_ON_ERROR);
+    }
+
+    /**
+     * 1. Fetch route for all stations of a trip at brouter
+     * 2. split the route by stations (we need the GeoJSON splitted!)
+     * 3. Create features for every station and route between
+     *
+     * @param HafasTrip $trip
+     *
+     * @return void
+     * @throws \JsonException
+     */
+    public static function reroutePolyline(HafasTrip $trip): void {
+        //1. Prepare coordinates from stations
+        $coordinates = [];
+        foreach ($trip->stopoversNEW as $stopover) {
+            $coordinates[] = [
+                $stopover->trainStation->latitude,
+                $stopover->trainStation->longitude,
+            ];
+        }
+
+        //2. Request route at brouter
+        $brouterGeoJSON = self::getGeoJSONForRoute($coordinates);
+        //remove unnecessary data
+        unset($brouterGeoJSON->features[0]->properties->messages, $brouterGeoJSON->features[0]->properties->times);
+
+        //3. Create "new" GeoJSON splitted by stations (as features)
+        $geoJson = ['type' => 'FeatureCollection', 'features' => []];
+
+        foreach ($brouterGeoJSON?->features[0]?->geometry?->coordinates ?? [] as $coordinate) {
+            $geoJson['features'][] = [
+                'type'       => 'feature',
+                'properties' => [],
+                'geometry'   => [
+                    'type'        => 'Point',
+                    'coordinates' => [
+                        $coordinate[0],
+                        $coordinate[1],
+                    ]
+                ]
+            ];
+        }
+
+        //4. Try to map stations to GeoJSON
+        $highestMappedKey = null;
+        foreach ($trip->stopoversNEW as $stopover) {
```

Comment:
Please create an issue for this. This is already technical debt, cause these calculations are very expensive. We need to optimize this as soon as we know how this could be done. ^^'